test/oas2-request-header-date-correct-type.spec.js

```diff
@@ -39,7 +39,7 @@ describe(ruleName, () => {
     expect(res).toEqual([
       {
         code: ruleName,
-        message: 'Date header should be type \'string\' and should not use the built-in OpenAPI format. Instead, \'pattern\' should be used to specify a custom format (https://developer.cisco.com/docs/api-insights/#!api-guidelines-analyzer)',
+        message: 'HTTP headers follow the syntax specified in the corresponding RFCs.; "[0].pattern" property must be truthy',
```

Comment:
Take this for example. If the message is not good. we can change the rule's description in api-insights-rulesets.js for "oas2-request-heaer-date-correct-type". to  something more meaningful. Here https://github.com/cisco-developer/api-insights-openapi-rulesets/blob/am-update-analyzer-message/api-insights-openapi-ruleset.js#L698

Reply:
As per discussions with @npateriya & @coliu19, selectively updating message formats to address the concerns laid out here -- https://github.com/cisco-developer/api-insights-openapi-rulesets/pull/20#issuecomment-1666236733